react/components/ProductAssemblyOptions/ProductAssemblyOptionsGroup.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { Fragment, FC } from 'react'
+import { propEq } from 'ramda'
+
+import ProductAssemblyItem from './ProductAssemblyItem'
+
+const maxIsOne = propEq('maxQuantity', 1)
+
+const getGroupType = (assemblyOption: AssemblyOptionGroup) => {
+  if (assemblyOption.maxQuantity === 1) {
+    return 'SINGLE'
+  }
+
+  if (Object.values(assemblyOption.items).every(maxIsOne)) {
+    return 'TOGGLE'
+  }
+
+  return 'MULTIPLE'
+}
+
+const getGroupPath = (assemblyTreePath?: TreePath[]) => {
+  let groupPath = [] as string[]
+  const treePath = assemblyTreePath || []
+  for (const currentPath of treePath) {
+    groupPath = groupPath.concat([
+      'items',
```

Comment:
now I get this, but the logic behind this groupPath should be documented somewhere